ark/phenotyping/som_utils.py

```diff
@@ -0,0 +1,115 @@
+import os
+import multiprocessing
+import subprocess
+
+import numpy as np
+import pandas as pd
+import xarray as xr
+import scipy.ndimage as ndimage
+from skimage.io import imread
+
+import ark.settings as settings
+from ark.utils import load_utils
+from ark.utils import misc_utils
+
+
+def create_pixel_matrix(img_xr, seg_labels, fovs=None, channels=None, blur_factor=2):
+    """Preprocess the images for FlowSOM clustering and creates a pixel-level matrix
+
+    Args:
+        img_xr (xarray.DataArray):
+            Array representing image data for each fov
+        seg_labels (xarray.DataArray):
+            Array representing segmentation labels for each image
+        fovs (list):
+            List of fovs to subset over, if None selects all
+        channels (list):
+            List of channels to subset over, if None selects all
```

Comment:
don't think this needs to be addressed now, but we should probably settle on a standard way of dealing with channels ('include_channels' vs 'exclude_channels' vs 'channels (None => all)' etc.).

Reply:
Definitely, we can handle that in a later PR.